src/plugins/data-card/use-data-card-tile-height.ts

```diff
@@ -0,0 +1,44 @@
+import { useEffect, useRef } from "react";
+import { kButtonSpace, kExampleDeckHeight } from "./data-card-types";
+
+interface UseDataCardTileHeightProps {
+  tileElt: HTMLElement | null;
+  height: number;
+  currEditAttrId: string;
+  modelId: string;
+  documentId: string | undefined;
+  readOnly: boolean;
+  onRequestRowHeight: (id: string, height: number) => void;
+  attrCount: number;
+}
+
+export const useDataCardTileHeight = ({
+  tileElt,
+  height,
+  currEditAttrId,
+  modelId,
+  documentId,
+  readOnly,
+  onRequestRowHeight,
+  attrCount
+}: UseDataCardTileHeightProps) => {
+
+  const attrCtRef = useRef(attrCount);
+
+  useEffect(() => {
+    if (!tileElt) return;
+    const uiHeight = tileElt.querySelector(".data-card-container")?.scrollHeight || 0;
+    const spaceLeft = height ? height - uiHeight : 0;
+    const requiringHeightChange = attrCtRef.current < attrCount || spaceLeft < kButtonSpace;
+    const loadingInContent = !documentId && readOnly;
```

Comment:
I'd recommend adding a comment about how the documentId will be undefined when the tile is being rendered in the curriculum tabs.